src/managers/cluster.py

```diff
@@ -26,3 +37,93 @@ def get_host_mapping(self) -> dict[str, str]:
         ip = socket.gethostbyname(hostname)
 
         return {"hostname": hostname, "ip": ip}
+
+    def get_leader(self) -> str:
+        """Query the etcd cluster for the raft leader and return the client_url as string."""
+        leader = ""
+
+        # loop through list of hosts and compare their member id with the leader
+        # if they match, return this host's endpoint
+        for endpoint in self.cluster_endpoints:
+            client = EtcdClient(client_url=endpoint)
+            try:
+                endpoint_status = client.get_endpoint_status()
+                member_id = endpoint_status["Status"]["header"]["member_id"]
+                leader_id = endpoint_status["Status"]["leader"]
+                if member_id == leader_id:
+                    leader = endpoint
+                    break
+            except KeyError:
+                # for now, we don't raise an error if there is no leader
+                # this may change when we have actual relevant tasks performed against the leader
+                logger.warning("No raft leader found in cluster.")
+
+        return leader
+
+
+class EtcdClient:
```

Comment:
Is there a best practise on where we can store these `helper` classes. I would prefer to have them in independent files if possible unless it's a DP best standard.

Reply:
For example I will use the class for TLS as well and I think it will be used in multiple managers.

Reply:
I've now moved `EtcdClient` to `src/common/client.py`. This way we can easily use it in multiple managers and don't have to handle it as a lib. The path `src/common/` follows the example from several Big Data repos (e.g. Spark, Kyuubi).